packages/mobile/src/harmony-native/components/TextLink/InternalLink.tsx

```diff
@@ -0,0 +1,80 @@
+import type { ReactNode } from 'react'
+import { useCallback } from 'react'
+
+import { getPathFromAudiusUrl } from '@audius/common/utils'
+import type { NavigationAction } from '@react-navigation/native'
+import { useLinkProps, useLinkTo } from '@react-navigation/native'
+import type { To } from '@react-navigation/native/lib/typescript/src/useLinkTo'
+import type { GestureResponderEvent } from 'react-native'
+
+import type { GestureResponderHandler } from 'app/types/gesture'
+
+import { TextPressable } from './TextPressable'
+
+export type InternalLinkToProps<
+  ParamList extends ReactNavigation.RootParamList
+> = {
+  to: To<ParamList>
+  action?: NavigationAction
+  target?: string
+  onPress?: (e: GestureResponderEvent) => void
+  children?: ReactNode
+}
+
+export const InternalLinkTo = <ParamList extends ReactNavigation.RootParamList>(
```

Comment:
the difference here is one is with a full url and the other would be with an internal route?

Reply:
Yup! It's the difference between passing a `url` prop vs a `to` prop

Reply:
got it. At first glance i feel like this could be one component, but probably cleaner as two separate components. good stuff 👍